src/auth/auth.ts

```diff
@@ -527,8 +526,12 @@ authStore.subscribe(
   withErrorIgnoring(async (state: AuthState, oldState: AuthState) => {
     if (!oldState.termsOfService.permitsSystemUsage && state.termsOfService.permitsSystemUsage) {
       if (window.Appcues) {
-        window.Appcues.identify(userStore.get().terraUser.id!, {
-          dateJoined: parseJSON((await Ajax().User.firstTimestamp()).timestamp).getTime(),
+        const { terraUser, samUser } = userStore.get();
+        // for Sam users who registered before we started tracking registration dates in Sam,
+        // registeredAt may be null in the Sam db. In that case, use epoch instead
+        const dateJoined = samUser.registeredAt ? samUser.registeredAt.getTime() : new Date('1970-01-01').getTime();
```

Comment:
I don't think we want to user `createdAt` because then a user who is invited, waits a month to register, then comes to Terra would immediately be shown the survey